src/mgm/tdes.rs

```diff
@@ -1,70 +0,0 @@
-use zeroize::Zeroizing;
-
-/// Size of a DES key
-pub(super) const DES_LEN_DES: usize = 8;
-
-/// Size of a 3DES key
-pub(crate) const DES_LEN_3DES: usize = DES_LEN_DES * 3;
-
-/// Weak and semi weak DES keys as taken from:
-/// %A D.W. Davies
-/// %A W.L. Price
-/// %T Security for Computer Networks
-/// %I John Wiley & Sons
-/// %D 1984
-const WEAK_DES_KEYS: &[[u8; DES_LEN_DES]] = &[
-    // weak keys
-    [0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01],
-    [0xFE, 0xFE, 0xFE, 0xFE, 0xFE, 0xFE, 0xFE, 0xFE],
-    [0x1F, 0x1F, 0x1F, 0x1F, 0x0E, 0x0E, 0x0E, 0x0E],
-    [0xE0, 0xE0, 0xE0, 0xE0, 0xF1, 0xF1, 0xF1, 0xF1],
-    // semi-weak keys
-    [0x01, 0xFE, 0x01, 0xFE, 0x01, 0xFE, 0x01, 0xFE],
-    [0xFE, 0x01, 0xFE, 0x01, 0xFE, 0x01, 0xFE, 0x01],
-    [0x1F, 0xE0, 0x1F, 0xE0, 0x0E, 0xF1, 0x0E, 0xF1],
-    [0xE0, 0x1F, 0xE0, 0x1F, 0xF1, 0x0E, 0xF1, 0x0E],
-    [0x01, 0xE0, 0x01, 0xE0, 0x01, 0xF1, 0x01, 0xF1],
-    [0xE0, 0x01, 0xE0, 0x01, 0xF1, 0x01, 0xF1, 0x01],
-    [0x1F, 0xFE, 0x1F, 0xFE, 0x0E, 0xFE, 0x0E, 0xFE],
-    [0xFE, 0x1F, 0xFE, 0x1F, 0xFE, 0x0E, 0xFE, 0x0E],
-    [0x01, 0x1F, 0x01, 0x1F, 0x01, 0x0E, 0x01, 0x0E],
-    [0x1F, 0x01, 0x1F, 0x01, 0x0E, 0x01, 0x0E, 0x01],
-    [0xE0, 0xFE, 0xE0, 0xFE, 0xF1, 0xFE, 0xF1, 0xFE],
-    [0xFE, 0xE0, 0xFE, 0xE0, 0xFE, 0xF1, 0xFE, 0xF1],
-];
-
-/// Is this 3DES key weak?
-///
-/// This check is performed automatically when the key is instantiated to
-/// ensure no such keys are used.
-pub(super) fn is_weak_key(key: &[u8; DES_LEN_3DES]) -> bool {
```

Comment:
Hmm, looking at this it's not quite equivalent to what we have upstream:

https://github.com/RustCrypto/block-ciphers/blob/master/des/src/tdes.rs#L32-L50

I should probably save switching for a separate PR.

Reply:
Opened #621